adhocracy4/polls/assets/PollDashboard/EditPollManagement.jsx

```diff
@@ -0,0 +1,260 @@
+        [position, 0, questions[index]] // insert to new index
+      ]
+    }
+    setQuestions(update(questions, diff))
+  }
+
+  const handleQuestionMoveDown = (index) => {
+    let diff = {}
+    const position = index + 1
+    diff = { $splice: [[index, 1], [position, 0, questions[index]]] }
+    setQuestions(update(questions, diff))
+  }
+
+  // | Choice state related handlers
+
+  const getNewChoice = (label = '', isOther = false) => {
+    return {
+      label,
+      key: isOther ? 'other-choice' : getNextLocalKey(),
+      is_other_choice: isOther
+    }
+  }
+
+  const handleChoiceLabel = (index, choiceIndex, label) => {
+    const diff = {}
+    diff[index] = { choices: {} }
+    diff[index].choices[choiceIndex] = { $merge: { label } }
+    setQuestions(update(questions, diff))
+  }
+
+  const handleChoiceAppend = (index, hasOtherOption) => {
+    const position = questions[index].choices.length - 1
+    const newChoice = getNewChoice()
+    const diff = {}
+    diff[index] = hasOtherOption
+      ? { choices: { $splice: [[position, 0, newChoice]] } }
+      : { choices: { $push: [newChoice] } }
+    setQuestions(update(questions, diff))
+  }
+
+  const handleChoiceIsOtherChoice = (index, isOtherChoice) => {
+    const diff = {}
+    if (isOtherChoice) {
+      const otherChoice = getNewChoice('other', true)
+      diff[index] = { choices: { $push: [otherChoice] } }
+    } else {
+      const choiceIndex = questions[index].choices.findIndex(c => c.key === 'other-choice')
+      diff[index] = { choices: { $splice: [[choiceIndex, 1]] } }
+    }
+    setQuestions(update(questions, diff))
+  }
+
+  const handleChoiceDelete = (index, choiceIndex) => {
+    const diff = {}
+    diff[index] = { choices: { $splice: [[choiceIndex, 1]] } }
+    setQuestions(update(questions, diff))
+  }
+
+  // | Poll form and submit logic
+
+  const removeAlert = () => {
+    setAlert(null)
+  }
+
+  const handleSubmit = (e) => {
+    console.log(questions)
+    e.preventDefault()
+
+    const data = {
+      questions
+    }
+    console.log(data.questions)
+
+    api.poll.change(data, props.pollId)
+      .done((data) => {
+        console.log(data.questions)
+        setQuestions(data.questions)
+        setAlert({
+          type: 'success',
+          message: django.gettext('The poll has been updated.')
+        })
+        setErrors([])
+        if (props.reloadOnSuccess) {
+          dashboard.updateDashboard()
+        }
+      })
+      .fail((xhr, status, err) => {
+        if (xhr.responseJSON && 'questions' in xhr.responseJSON) {
+          setErrors(xhr.responseJSON.questions)
+        }
+
+        setAlert({
+          type: 'danger',
+          message: django.gettext('The poll could not be updated.')
+        })
+      })
+  }
+
+  // | Runtime logic and JSX render
+
+  const [questions, setQuestions] = useState([])
+  const [errors, setErrors] = useState([])
+  const [alert, setAlert] = useState(null)
```

Comment:
mostly a personal pref but I think those should always be at the very top of the component